dbms/src/Functions/FunctionsHashing.h

```diff
@@ -353,6 +353,43 @@ struct JavaHashImpl
     static constexpr bool use_int_hash_for_pods = false;
 };
 
+struct JavaHashUTF16LEImpl
+{
+    static constexpr auto name = "javaHashUTF16LE";
+    using ReturnType = Int32;
+
+    static Int32 apply(const char * raw_data, const size_t raw_size)
+    {
+        char * data = const_cast<char *>(raw_data);
+        size_t size = raw_size;
+
+        // Remove Byte-order-mark(0xFFFE) for UTF-16LE
+        if (size >= 2 && data[0] == -1 && data[1] == -2)
```

Comment:
You compare `char` with `int`. And char may be signed or unsigned.
Either replace `char` to `Int8` or `UInt8`
or replace -1 to `'\xFF'` and -2 to `'\xFE'`.